.cirrus.yml

```diff
@@ -46,29 +53,40 @@ task:
     - brew install zstd
     # Ensure user have access to network devices instead of giving super-user right
     - sudo chmod a+rw /dev/bpf*
-    - cmake -DLIGHT_PCAPNG_ZSTD=ON -S . -B Dist
+    - cmake -DLIGHT_PCAPNG_ZSTD=ON -DPCAPPP_BUILD_COVERAGE=ON -S . -B Dist
   build_script:
     - cmake --build Dist -j 2
-  install_tcpreplay_script:
-    - brew install tcpreplay
+  install_tcpreplay_gcovr_script:
+    - brew install tcpreplay gcovr
   test_pcapplusplus_script:
     - python3 -m ensurepip
     - python3 -m pip install -r ci/run_tests/requirements.txt
     - python3 ci/run_tests/run_tests.py --interface en0 --test-args="-x TestPcapLiveDeviceSpecialCfg"
+  coverage_report_script:
+    - gcovr -v -r . $GCOVR_FLAGS -o coverage.xml
+    - curl -Os https://uploader.codecov.io/latest/macos/codecov
+    - chmod +x codecov
+    - ./codecov -v -f coverage.xml -F zstd -F macos-ventura -F unittest || true
 
 task:
   name: Regression Tests Linux (Address Sanitizer)
   container:
     image: gcr.io/oss-fuzz-base/base-clang
   configure_script:
-    - apt-get update && apt-get install -y cmake git libpcap-dev
-    - cmake -S . -B $BUILD_DIR -DPCAPPP_USE_SANITIZER=AddressSanitizer -DPCAPPP_BUILD_FUZZERS=ON -DPCAPPP_BUILD_EXAMPLES=OFF
+    - apt-get update && apt-get install -y cmake git libpcap-dev gcovr curl
```

Comment:
Do we need to run `gcovr` in Address Sanitizer tests? 🤔

Reply:
It is an option to see where current regression tests touch. Not affect coverage much, just for visualization. It can be removed if you want.

Reply:
I don't think these tests cover code that is not already covered in other tests, am I wrong?

Reply:
Not check all of them but yes, they are already covered. I'll remove